content/docs/hoon/guides/json-guide.md

```diff
@@ -0,0 +1,453 @@
+  }
+]
+```
+
+Let's also see how that `$json` would look as real JSON encoded in text. We can do that with `+en-json:html`:
+
+```
+> (crip (en-json:html (to-js:user-lib usr)))
+'{"joined":1631440078,"username":"john456","name":["John","William","Smith"],"email":"john.smith@example.com"}'
+```
+
+Finally, let's try converting the `$json` back to a `$user` again with our `+from-js` arm:
+
+```
+> (from-js:user-lib usr-json)
+[ username='john456'
+  name=[first='John' mid='William' last='Smith']
+  joined=~2021.9.12..09.47.58
+  email='john.smith@example.com'
+]
+```
+
+### Analysis
+
+#### Converting to `$json`
+
+Here's our arm that converts a `$user` structure to `$json`:
+
+```hoon
+++  to-js
+  |=  usr=user
+  |^  ^-  json
+  %-  pairs:enjs:format
+  :~
+    ['username' s+username.usr]
+    ['name' name]
+    ['joined' (sect:enjs:format joined.usr)]
+    ['email' s+email.usr]
+  ==
+  ++  name
+    :-  %a
+    :~
+      [%s first.name.usr]
+      [%s mid.name.usr]
+      [%s last.name.usr]
+    ==
+  --
+```
+
+There are different ways we could represent our `$user` structure as JSON, but in this case we've opted to encapsulate it in an object and have the `name` as an array (since JSON arrays preserve order).
+
+[`+enjs:format`](/docs/hoon/reference/zuse/2d_1-5#enjsformat)includes the convenient [`+pairs`](/docs/hoon/reference/zuse/2d_1-5#pairsenjsformat) function, which converts a list of `[@t json]` to an object containing those key-value pairs. We've used this to assemble the final object. Note that if you happen to have only a single key-value pair rather than a list, you can use [`+frond`](/docs/hoon/reference/zuse/2d_1-5#frondenjsformat) instead of `+pairs`.
+
+For the `joined` field, we've used the [`+sect`](/docs/hoon/reference/zuse/2d_1-5#sectenjsformat) function from `+enjs` to convert the `@da` to a Unix seconds timestamp in a `$json` number. The `+sect` function, like others in `+enjs`, takes in a noun (in this case a `@da`) and produces `$json` (in this case a `[%n @ta]` number). `+enjs` contains a handful of useful functions like this, but for the rest we've just hand-made the `$json` structure. This is fairly typical when encoding `$json`, it's usually [decoding](#converting-from-json) that makes more extensive use of the `$json` utility functions in `+format`.
+
+For the `name` field we've just formed a cell of `%a` and a list of `$json` strings, since a `$json` array is `[%a p=(list json)]`. Note we've separated this part into its own arm and wrapped the whole thing in a `|^` - a core with a `$` arm that's computed immediately. This is simply for readability - our structure here is quite simple but when dealing with deeply-nested `$json` structures or complex logic, having a single giant function can quickly become unwieldy.
+
+#### Converting from `$json`
+
+Here's our arm that converts `$json` to our `$user` structure:
+
+```hoon
+++  from-js
+  =,  dejs:format
+  |=  jon=json
+  ^-  user
+  %.  jon
```

Comment:
This is not strictly necessary, cause these runes are effectively a no-op. I'd scrap these three lines and replace it with  `^-  $-(json user)` cast, to make it clear that the `dejs` reparsers are higher order functions